### DISCUSSION TITLE
Respond to user search events

### DISCUSSION THREAD

**Discussion by StarfallProjects** [squidfunk/mkdocs-material#5732] [Q&A] [answered]
I'm setting up Plausible Analytics in our docs. I'd like to use their [Events API](https://plausible.io/docs/events-api) to track search queries and results. Ideally, I'd log an event in Plausible at two points:

- When a user enters a search term (this would log what they searched and the results they got)
- When a user clicks on a result (this would log what they searched and the page they chose)

Has anyone done similar? I'm particularly stumped on the first one, as the user doesn't have to click anything to search (results appear as they type), so I'm not sure what I could hook into that signals they're done writing their query.

**Comment by squidfunk** [comment, marked as answer]
Search is your friend 😊

- #4847
- #5063

> Has anyone done similar? I'm particularly stumped on the first one, as the user doesn't have to click anything to search (results appear as they type), so I'm not sure what I could hook into that signals they're done writing their query.

Material for MkDocs uses the `blur` event to capture the semantics of "finished typing". Not perfect, but works reasonably well.

https://github.com/squidfunk/mkdocs-material/blob/c2e55585fa4ab4657d8c47ca51e83caee550f962/src/partials/integrations/analytics/google.html#L42-L49
